src/audio/component.c

```diff
@@ -57,17 +57,49 @@ static const struct comp_driver *get_drv(uint32_t type)
 	return drv;
 }
 
+static const struct comp_driver *get_drv_from_uuid(uint8_t *uuid)
+{
+	struct comp_driver_list *drivers = comp_drivers_get();
+	struct list_item *clist;
+	const struct comp_driver *drv = NULL;
+	struct comp_driver_info *info;
+	uint32_t flags;
+
+	irq_local_disable(flags);
+
+	/* search driver list for driver type */
+	list_for_item(clist, &drivers->list) {
+		info = container_of(clist, struct comp_driver_info, list);
+		if (!rstrncmp((const char *)info->drv->uid,
+			      (const char *)uuid, UUID_SIZE)) {
+			tr_info(&comp_tr,
+				"get_drv_from_uuid(), found driver type %d, uuid %s",
+				info->drv->type, info->drv->uid);
```

Comment:
There is some mismatch - UUID_SIZE is equal to 16, what is binary UUID number size, so it should be compared with `memcmp` and printing this value can't be done with `%s`, or when UUID should be passed in text form, then drv->uid size is too small.

Reply:
The uuid is passed in binary mode for the comparing, the printing here is for the full text(including the naming).

Reply:
why are we comparing strings ? I would expect us to compare the binary a.b.c.d data/struct

Reply:
changed now.